### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by SunPj**
Library && plugins updates

Scala 2.11.8 -> 2.12.12
Spark 2.2.0 -> 3.0.1

SBT plugins

sbt-microsites 0.4.0 -> 1.1.5
sbt-ghpages 0.5.4 -> 0.6.3
sbt-git 0.8.5-> 1.0.0
sbt-header 1.7.0 -> 5.6.0

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -9,54 +9,71 @@ resolvers += "swoop-bintray" at "https://dl.bintray.com/swoop-inc/maven/"
...
+  micrositeAuthor := "Swoop",
+  micrositeHomepage := "http://www.swoop.com",
+  micrositeBaseUrl := "spark-records",
+  micrositeDocumentationUrl := "/spark-records/docs.html",
+  micrositeGithubOwner := "swoop-inc",
+  micrositeGithubRepo := "spark-records",
+  micrositeHighlightTheme := "tomorrow"
+)
+
+
```
Comment (ssimeonov) [inline build.sbt:79]: Remove one line. :)

File: src/main/scala/com/swoop/spark/records/package.scala [hunk 2]
```diff
@@ -94,7 +94,7 @@ package object records {
       (implicit env: RecordEnvironment): DataFrame =
       unknownErrorDetailCounts(RootCauseAnalysis.classNameContains(classNameFragment)).drop("id_messages")
 
-    def recordData(implicit env: RecordEnvironment): Dataset[A] =
+    def recordData(implicit env: RecordEnvironment, enc: Encoder[A]): Dataset[A] =
```
Comment (ssimeonov) [inline src/main/scala/com/swoop/spark/records/package.scala:97]: Good change; narrows the implicit requirement.

File: src/main/scala/com/swoop/spark/records/package.scala [hunk 3]
```diff
@@ -32,14 +32,14 @@ package object records {
...
-  implicit class RecordsRichDataset[A <: Product : Encoder, Rec](val underlying: Dataset[Rec])
+  implicit class RecordsRichDataset[A <: Product, Rec](val underlying: Dataset[Rec])
     (implicit ev: Rec <:< Record[A, _]) extends Serializable {
 
     def errorRecords(implicit env: RecordEnvironment): Dataset[Rec] =
       env.errorRecords(underlying)
 
     def allIssues(implicit env: RecordEnvironment): DataFrame =
-      new RootCauseAnalysis(underlying.filter(_.issues.isDefined).toDF(), env).issues
+      new RootCauseAnalysis(underlying.filter((r: Rec) => r.issues.isDefined).toDF(), env).issues
```
Comment (ssimeonov) [inline src/main/scala/com/swoop/spark/records/package.scala:42]: Do you get a compiler error without the refactoring?

**Review by ssimeonov — APPROVED** [review]
Looks good. Minor style comments.

File: src/main/scala/com/swoop/spark/records/package.scala (hunk 3 above)
Comment (SunPj) [reply src/main/scala/com/swoop/spark/records/package.scala:42]: @ssimeonov Yes I did get a compiler error asking me to implicitly define the type. Seems that the new overloaded methods were introduced in last Spark version so this one became polymorphic, so we have to be explicit on this.

**Review by ssimeonov — APPROVED** [review]

**Pull request merged**